coreos-installer

```diff
@@ -317,6 +317,25 @@ write_ignition_file() {
     cp /tmp/ignition.ign /mnt/boot_partition/ignition/config.ign
 }
 
+############################################################
+# Helper to persist networking options to initramfs first boot
+############################################################
+write_networking_opts() {
+    if [ ! -f /tmp/ip_opt ]; then
+        return
+    fi
+
+    dialog --title 'CoreOS Installer' --infobox "Embedding provided networking options" 5 70
+    # check for the boot partition
+    mkdir -p /mnt/boot_partition
+    # TODO check to make sure the disk with label 'boot'
+    #      is part of ${DEST_DEV}
+    mount /dev/disk/by-label/boot /mnt/boot_partition
+    trap 'umount /mnt/boot_partition' RETURN
+
+    cp /tmp/ip_opt /mnt/boot_partition/ignition.firstboot
```

Comment:
if we stick with the implementation in https://github.com/coreos/ignition-dracut/pull/65 then i think we'd need to make the format of the file be `set ignition_network_kcmdline="$(cat /tmp/ip_opt)"`. I chose to make the file able to be sourced so that if we find other uses for it we can use other variable names in the future.

Also maybe we should append to the file rather than overwriting it completely.

Reply:
Is this file currently used for anything? I explicitly just overwrote it because my understanding was that it was currently only checked to exist rather than using it's contents.

Reply:
> Is this file currently used for anything? 

nope

> I explicitly just overwrote it because my understanding was that it was currently only checked to exist rather than using it's contents.

That is a correct assumption and this code as is works fine. My point is that we may find more uses for this file in the future and we can enable that now by using variables within the file and appending rather than entire file contents.

Reply:
Fair enough, I'll update it to append.